### PR TITLE
Add a check on PR in github workflows for labels and milestones

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Add a new check workflow and custom action to verify PR labels and milestone before merging into master.

There still is a limitation on github. We can't trigger the check on milestone change on the pr because the event is missing. 

I will comment that part until it's supported by github

### How to test it?

change the labels on the PR

### CODE REVIEW

File: .github/actions/check-pr-status/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const BLOCKING_LABELS = [`flag: 💥 Breaking change`, `flag: don't merge`];
+
+async function main() {
+  try {
+    const labels = context.payload.pull_request?.labels;
+
+    const hasBlockingLabel = labels.some(label => BLOCKING_LABELS.includes(label.name));
+    if (hasBlockingLabel) {
+      core.setFailed(
+        `The PR has been labelled with a blocking label (${BLOCKING_LABELS.join(', ')}).`
```
Comment (Convly) [inline .github/actions/check-pr-status/index.js:13]: Idea: Even if the list won't grow that much, it could maybe be more helpful to log the blocking labels that are actually set on this PR? This way you can see at a glance what's causing the issue.

File: .github/actions/check-pr-status/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "check-pr-status",
+  "version": "1.0.0",
+  "main": "dist/index.js",
+  "license": "MIT",
+  "private": true,
+  "scripts": {
+    "build": "NODE_ENV=production ncc build index.js -o dist --minify"
```
Comment (Convly) [inline .github/actions/check-pr-status/package.json:8]: Just wondering why is this needed? Can't just pass our `index` to the action? Is it just to beneficiate from the minifier?

**Review by Convly — CHANGES_REQUESTED** [review]
Very nice idea :fire:
Some questions/comments :slightly_smiling_face:

File: .github/actions/check-pr-status/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+const core = require('@actions/core');
+const { context } = require('@actions/github');
+
+const BLOCKING_LABELS = [`flag: 💥 Breaking change`, `flag: don't merge`];
+
+async function main() {
+  try {
+    const labels = context.payload.pull_request?.labels;
```
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply .github/actions/check-pr-status/index.js:8]: labels will always be at least an empty array if pull_request is defined

File: .github/actions/check-pr-status/package.json (hunk 2 above)
Comment (alexandrebodin) [reply .github/actions/check-pr-status/package.json:8]: Actually it's because github actions don't install the dependencies so I have to build a dist file that contains the deps and commit it

File: .github/actions/check-pr-status/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Convly) [reply .github/actions/check-pr-status/index.js:8]: Yeah but if `pull_request` is undefined, then labels will be too, that's what I was talking about

File: .github/actions/check-pr-status/package.json (hunk 2 above)
Comment (Convly) [reply .github/actions/check-pr-status/package.json:8]: Alright, that's nice to know, maybe we could put that in the readme? (like, how to update/contribute?)

**Review by soupette — APPROVED** [review]

**Review by Convly — APPROVED** [review]
lgtm
